api/envoy/extensions/filters/http/api_key_auth/v3/api_key_auth.proto

```diff
@@ -0,0 +1,69 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.http.api_key_auth.v3;
+
+import "envoy/config/core/v3/base.proto";
+
+import "udpa/annotations/sensitive.proto";
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.http.api_key_auth.v3";
+option java_outer_classname = "ApiKeyAuthProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/http/api_key_auth/v3;api_key_authv3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: APIKey Auth]
+// APIKey Auth :ref:`configuration overview <config_http_filters_api_key_auth>`.
+// [#extension: envoy.filters.http.api_key_auth]
+
+// Basic HTTP authentication.
+//
+// Example:
+//
+// .. code-block:: yaml
+//
+//    location: HEADER
+//    name: X-API-KEY
+//    keys:
+//      inline_string: |-
+//        apikey1:user1
+//        apikey2:user2
+//
+message APIKeyAuth {
+  enum APIKeyLocation {
+    // The API key is expected to be in a header.
+    HEADER = 0;
+
+    // The API key is expected to be in a query parameter.
+    QUERY = 1;
+
+    // The API key is expected to be in a cookie.
+    COOKIE = 2;
+  }
+
+  // The location of the API key.
+  APIKeyLocation location = 1;
```

Comment:
What is the header name or query name?

Reply:
If it's defined like this, 

```yaml
    location: HEADER
    name: X-API-KEY
```

the key comes in headers like `X-API-Key: asdf`.

If it's defined like this, 

```yaml
    location: QUERY
    name: api-key
```

the key comes in the query parameters like `example.com/foo/bar?api-key=asdf`.

Reply:
I personally will prefer to define this to three different string field like `authentication_header`, `authentication_query`, `authentication_cookie`. Then single field could be used to indicate both the value and the position.

And I hope the API could be consistant with the basic auth.

Reply:
Moved the location stuff into `oneof`.